## Makayla McKibben
### DSC550
### 09.01.2024
### Ex. 1.2

In [14]:
import pandas as pd
import numpy as np

In [15]:
# Load as a pandas data frame
df = pd.read_csv("C:/Users/makay/OneDrive/Documents/DSC550/Video_Games_Sales_as_at_22_Dec_2016.csv")

In [16]:
# Display the first ten rows of data
df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
0          76.0          51.0          8       322.0  Nintendo      E  
1           NaN           NaN        NaN         NaN       NaN    NaN  
2          82.0          73.0        8.3       709.0  Nintendo      E  
3          80.0          73.0          8       192.0  Nintendo      E  
4           NaN           NaN        NaN         NaN       NaN    NaN  
5           NaN           NaN        NaN         NaN       NaN    NaN  
6          89.0          65.0        8.5       431.0  Nintendo      E  
7          58.0          41.0        6.6       129.0  Nintendo      E  
8          87.0          80.0        8.4       594.0  Nintendo      E  
9           NaN           NaN        NaN         NaN       NaN    NaN

In [17]:
# Find the dimensions of the data frame
df.shape

(16719, 16)

##### This dataframe has 16719 rows and 16 columns. That means there are 16719 instances that have data which populate the columns.

In [19]:
# Find the top five games by critic score
df.sort_values(by = ["Critic_Score"], ascending = False).head(5)

Name Platform  Year_of_Release     Genre  \
227   Tony Hawk's Pro Skater 2       PS           2000.0    Sports   
57         Grand Theft Auto IV      PS3           2008.0    Action   
51         Grand Theft Auto IV     X360           2008.0    Action   
5350               SoulCalibur       DC           1999.0  Fighting   
165         Grand Theft Auto V     XOne           2014.0    Action   

                 Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
227             Activision      3.05      1.41      0.02         0.20   
57    Take-Two Interactive      4.76      3.69      0.44         1.61   
51    Take-Two Interactive      6.76      3.07      0.14         1.03   
5350    Namco Bandai Games      0.00      0.00      0.34         0.00   
165   Take-Two Interactive      2.81      2.19      0.00         0.47   

      Global_Sales  Critic_Score  Critic_Count User_Score  User_Count  \
227           4.68          98.0          19.0        7.7       299.0   
57           10.50          98.0          64.0        7.5      2833.0   
51           11.01          98.0          86.0        7.9      2951.0   
5350          0.34          98.0          24.0        8.8       200.0   
165           5.48          97.0          14.0        7.9       764.0   

                    Developer Rating  
227   Neversoft Entertainment      T  
57             Rockstar North      M  
51             Rockstar North      M  
5350                    Namco      T  
165            Rockstar North      M

In [20]:
# Find the number of games in each genre
df['Genre'].value_counts()

Genre
Action          3370
Sports          2348
Misc            1750
Role-Playing    1500
Shooter         1323
Adventure       1303
Racing          1249
Platform         888
Simulation       874
Fighting         849
Strategy         683
Puzzle           580
Name: count, dtype: int64

In [21]:
# Find the first five games on the SNES platform
df[df.Platform == 'SNES'].sort_values("Year_of_Release").head(5)

Name Platform  Year_of_Release       Genre Publisher  \
18    Super Mario World     SNES           1990.0    Platform  Nintendo   
1195        Final Fight     SNES           1990.0      Action    Capcom   
511              F-Zero     SNES           1990.0      Racing  Nintendo   
1791         Pilotwings     SNES           1990.0  Simulation  Nintendo   
1111      Super Scope 6     SNES           1991.0     Shooter  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
18       12.78      3.75      3.54         0.55         20.61           NaN   
1195      0.67      0.17      0.69         0.03          1.56           NaN   
511       1.37      0.51      0.89         0.07          2.85           NaN   
1791      0.46      0.17      0.48         0.02          1.14           NaN   
1111      1.06      0.38      0.15         0.05          1.65           NaN   

      Critic_Count User_Score  User_Count Developer Rating  
18             NaN        NaN         NaN       NaN    NaN  
1195           NaN        NaN         NaN       NaN    NaN  
511            NaN        NaN         NaN       NaN    NaN  
1791           NaN        NaN         NaN       NaN    NaN  
1111           NaN        NaN         NaN       NaN    NaN

In [22]:
# Finds all the publishers
publishers = df["Publisher"].unique()

In [23]:
# Find the five publishers with the highest total global sales
z = []
for x in publishers:
    pub_sales = df[df.Publisher == x]["Global_Sales"].sum()
    print(x, " : ", pub_sales)
    z.append(x)
    z.append(pub_sales)
z    

Nintendo  :  1788.81
Microsoft Game Studios  :  248.32
Take-Two Interactive  :  403.82
Sony Computer Entertainment  :  606.48
Activision  :  731.16
Ubisoft  :  471.61
Bethesda Softworks  :  85.05
Electronic Arts  :  1116.96
Sega  :  270.34999999999997
SquareSoft  :  57.650000000000006
Atari  :  156.82999999999998
505 Games  :  55.32
Capcom  :  200.02
GT Interactive  :  25.23
Konami Digital Entertainment  :  282.39000000000004
Square Enix  :  144.35
Sony Computer Entertainment Europe  :  23.37
Virgin Interactive  :  43.87000000000001
LucasArts  :  85.83
Warner Bros. Interactive Entertainment  :  151.79
Universal Interactive  :  17.770000000000003
Eidos Interactive  :  98.65
RedOctane  :  8.68
Vivendi Games  :  57.940000000000005
Enix Corporation  :  33.739999999999995
Namco Bandai Games  :  254.62
Palcom  :  4.17
Hasbro Interactive  :  15.22
THQ  :  338.44000000000005
Fox Interactive  :  7.389999999999999
Acclaim Entertainment  :  66.57
MTV Games  :  20.720000000000002
Disney Interactiv

['Nintendo',
 1788.81,
 'Microsoft Game Studios',
 248.32,
 'Take-Two Interactive',
 403.82,
 'Sony Computer Entertainment',
 606.48,
 'Activision',
 731.16,
 'Ubisoft',
 471.61,
 'Bethesda Softworks',
 85.05,
 'Electronic Arts',
 1116.96,
 'Sega',
 270.34999999999997,
 'SquareSoft',
 57.650000000000006,
 'Atari',
 156.82999999999998,
 '505 Games',
 55.32,
 'Capcom',
 200.02,
 'GT Interactive',
 25.23,
 'Konami Digital Entertainment',
 282.39000000000004,
 'Square Enix',
 144.35,
 'Sony Computer Entertainment Europe',
 23.37,
 'Virgin Interactive',
 43.87000000000001,
 'LucasArts',
 85.83,
 'Warner Bros. Interactive Entertainment',
 151.79,
 'Universal Interactive',
 17.770000000000003,
 'Eidos Interactive',
 98.65,
 'RedOctane',
 8.68,
 'Vivendi Games',
 57.940000000000005,
 'Enix Corporation',
 33.739999999999995,
 'Namco Bandai Games',
 254.62,
 'Palcom',
 4.17,
 'Hasbro Interactive',
 15.22,
 'THQ',
 338.44000000000005,
 'Fox Interactive',
 7.389999999999999,
 'Acclaim Entertainmen

In [24]:
y = {z[i]: z[i+1] for i in range(0, len(z), 2)}
{k: v for k, v in sorted(y.items(), key = lambda item: item[1])}
# Last five values are the top five publishers of global sales

{nan: 0.0,
 'Lighthouse Interactive': 0.01,
 'Michaelsoft': 0.01,
 'Media Entertainment': 0.01,
 'Genterprise': 0.01,
 'Interworks Unlimited, Inc.': 0.01,
 'Inti Creates': 0.01,
 'Boost On': 0.01,
 'EON Digital Entertainment': 0.01,
 'Naxat Soft': 0.01,
 'Piacci': 0.01,
 'Paradox Development': 0.01,
 'Otomate': 0.01,
 'Ascaron Entertainment': 0.01,
 'Ongakukan': 0.01,
 'Commseed': 0.01,
 'UIG Entertainment': 0.01,
 'Takuyo': 0.01,
 'Interchannel-Holon': 0.01,
 'Red Flagship': 0.01,
 'GOA': 0.02,
 'King Records': 0.02,
 'Graphsim Entertainment': 0.02,
 'Gaga': 0.02,
 'Plenty': 0.02,
 'fonfun': 0.02,
 'Epic Games': 0.02,
 'Quintet': 0.02,
 'Phoenix Games': 0.02,
 'Marvelous Games': 0.02,
 'Codemasters Online': 0.02,
 'Imageworks': 0.02,
 'Karin Entertainment': 0.02,
 'Technos Japan Corporation': 0.02,
 'Masque Publishing': 0.02,
 'New World Computing': 0.02,
 'Mirai Shounen': 0.02,
 'Visco': 0.02,
 'Saurus': 0.02,
 'Revolution (Japan)': 0.02,
 'Giza10': 0.02,
 'Giga': 0.02,
 'Mycom': 0.0

In [25]:
# Create a new column in the data frame that calculates the percentage of sales from NA
df["NA_Percent_Sales"] = df["NA_Sales"]/df["Global_Sales"]*100
df.head(5)

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  NA_Percent_Sales  
0          51.0          8       322.0  Nintendo      E         50.115110  
1           NaN        NaN         NaN       NaN    NaN         72.266402  
2          73.0        8.3       709.0  Nintendo      E         44.144144  
3          73.0          8       192.0  Nintendo      E         47.635032  
4           NaN        NaN         NaN       NaN    NaN         35.926044

In [26]:
# Find the number of NaN entries in each column
namenan = df['Name'].isna().sum()
platnan = df['Platform'].isna().sum() 
yearnan = df['Year_of_Release'].isna().sum()
genrenan = df['Genre'].isna().sum()
pubnan = df['Publisher'].isna().sum()
NAnan = df['NA_Sales'].isna().sum()
EUnan = df['EU_Sales'].isna().sum()
JPnan = df['JP_Sales'].isna().sum()
othernan = df['Other_Sales'].isna().sum()
globalnan = df['Global_Sales'].isna().sum()
critscorenan = df['Critic_Score'].isna().sum()
critcountnan = df['Critic_Count'].isna().sum()
userscorenan = df['User_Score'].isna().sum()
usercountnan = df['User_Count'].isna().sum()
devnan = df['Developer'].isna().sum()
ratingnan = df['Rating'].isna().sum()
NApernan = df['NA_Percent_Sales'].isna().sum()

In [27]:
# Find median user score of all games
df['User_Score'] = df['User_Score'].apply(pd.to_numeric, errors = 'coerce')
df['User_Score'].median()

7.5

In [28]:
# Replace all NaN with the median
df['User_Score'].fillna(7.5)

0        8.0
1        7.5
2        8.3
3        8.0
4        7.5
        ... 
16714    7.5
16715    7.5
16716    7.5
16717    7.5
16718    7.5
Name: User_Score, Length: 16719, dtype: float64